src/lib/io/sol-gpio-impl-contiki.c

```diff
@@ -61,8 +61,10 @@ static void
 event_handler_cb(void *user_data, process_event_t ev, process_data_t ev_data)
 {
     struct sol_gpio *gpio = user_data;
+    bool val;
 
-    gpio->irq.cb(gpio->irq.data, gpio);
+    val = sol_gpio_read(gpio);
```

Comment:
On Fri, 13 Nov 2015 14:15:49 -0800, Gustavo Sverzut Barbieri wrote:
> >  
> > -    gpio->irq.cb(gpio->irq.data, gpio);
> > +    val = sol_gpio_read(gpio);
> 
> just to check, do we need to read it here? Can't we put that in the ISR and save the value in `ev` or `ev_data`?
> 

On Contiki it wouldn't make any difference, as the moment you get to
poll the ISR, it will immediately call the callback. And we don't even
know at poll time how to read it anyway, because we don't know what kind
of sensor it is.